picard/formats/id3.py

```diff
@@ -330,6 +333,20 @@ def _load(self, filename):
                 if frame.desc:
                     name += ':%s' % frame.desc
                 metadata.add(name, frame.text)
+            elif frameid == 'SYLT':
```

Comment:
IMO they are not synced lyrics unless also `frame.type== 1`. (See [SYLT](https://mutagen-specs.readthedocs.io/en/latest/id3/id3v2.4.0-frames.html#synchronised-lyrics-text).)

But if we are going to accept any SYLT type, then we probably need to preserve it since you might have multiple types.

We also absolutely need to preserve the lyrics language - because you might have several SYLT frames with different languages. How about naming the metadata `syncedlyrics:lang:desc`?

We may also need to handle (or raise an exception) if `frame.format == 0` (i.e. timestamp is in MPEG frames rather than millisecs).

I am unclear whether we need to preserve the `frame.encoding` value and/or e.g. convert the contents from ISO-8859-1 to UNICODE or vice-versa to standardise the metadata.

Reply:
> IMO they are not synced lyrics unless also frame.type== 1. (See SYLT.)
>
> But if we are going to accept any SYLT type, then we probably need to preserve it since you might have multiple types.

I think it should only read / write type 1 lyrics. Otherwise the becomes too complex for the tag name. Other types should use other tag names if needed.

> We also absolutely need to preserve the lyrics language - because you might have several SYLT frames with different languages. How about naming the metadata syncedlyrics:lang:desc?

Absolutely. And we should from the beginning start with a syntax that avoids ambiguity between syncedlyrcis:lang and syncedlyrics:description. As I proposed in PICARD-2844 I would use `syncedlyrics::description`  for the case where language is empty but description is set.

> I am unclear whether we need to preserve the frame.encoding value and/or e.g. convert the contents from ISO-8859-1 to UNICODE or vice-versa to standardise the metadata.

Encoding is handled transparently already.

Reply:
> Absolutely. And we should from the beginning start with a syntax that avoids ambiguity between syncedlyrcis:lang and syncedlyrics:description.

I followed your proposal, but have a small doubt. This makes specifying the language mandatory when accessing the metadata, or am I wrong? Or should `syncedlyrics` default to a specific language, like `syncedlyrics:eng`? Also, I made `eng` the default language like it already is for comments.

Reply:
No, it doesn't make specifying the language mandatory. We can still allow just `syncedlyrics` and default to `eng` for that.

The only thing it does is removing the ambiguity of `comment:xxx`. Is this a comment with language `abc` or a comment with description `abc`? In the proposal you have `comment:abc` (the language, with empty description) and `comment::abc` (the description, with default language).